sqlobject/src/test/java/org/jdbi/v3/docs/TestInClauseExpansion.java

```diff
@@ -63,9 +65,25 @@ public void testInClauseExpansion() throws Exception
     }
 
     @UseStringTemplate3StatementLocator
+    @RegisterContainerMapper(ImmutableSetCollectorFactory.class)
     public interface DAO
     {
         @SqlQuery
-        Set<String> findIdsForNames(@BindIn("names") List<Integer> names);
+        ImmutableSet<String> findIdsForNames(@BindIn("names") List<Integer> names);
+    }
+
+    public static class ImmutableSetCollectorFactory<T>
+            implements CollectorFactory<T, ImmutableSet.Builder<T>, ImmutableSet<T>> {
+
+        public boolean accepts(Class<?> type) {
+            return ImmutableSet.class.isAssignableFrom(type);
+        }
+
+        @Override
+        public Collector<T, ImmutableSet.Builder<T>, ImmutableSet<T>> newCollector(Class<ImmutableSet<T>> type) {
+            return Collector.of(ImmutableSet.Builder::new, ImmutableSet.Builder::add, (first, second) -> {
+                throw new UnsupportedOperationException("Parallel collecting is not supported");
+            }, ImmutableSet.Builder::build, Collector.Characteristics.UNORDERED);
+        }
```

Comment:
I think that is a great idea, and it could be part of the initial release.  It is pretty easy to add another module.